### ISSUE TITLE
YAML title in sidebar not displayed correctly

### ISSUE THREAD

**Issue by jsprklst** [mkdocs/mkdocs#1566]
When using YAML front matter to set the title (without a H1 title in the markdown file and a title in mkdocs.yml) the titles are rendered incorrectly, but only *after the selected document*.

* mkdocs 0.17.5
* python 3.6.1

sf_code_review.md:
```markdown
---
title: Code review
---

No H1 header...
```

mkdocs.yml:
```yml
pages:
  - Home: 
    - sf_installation.md
    - sf_advanced.md
    - sf_code_overview.md
    - sf_code_review.md
    - sf_code_search.md
```

Screenshots:
![image](https://user-images.githubusercontent.com/1721525/43207083-7fa86cf6-9027-11e8-8276-0126033be7c1.png)
![image](https://user-images.githubusercontent.com/1721525/43207096-860026b6-9027-11e8-9dfd-39b455aca50a.png)
![image](https://user-images.githubusercontent.com/1721525/43207110-8b9a77f2-9027-11e8-91fd-857f464c6b09.png)

**Comment by jsprklst** [comment]
Addition: the Lunr.js search only indexes the YAML title if there are no headers in the markdown.

**Comment by waylan** [comment]
This is a duplicate of #1347. Note that a fix is available in the 1.0 beta release (currently 1.0b1). 1.0 final should be available in a  few weeks.
